src/routing_membrane.rs

```diff
@@ -502,7 +496,8 @@ impl<F> RoutingMembrane<F> where F: Interface {
         }
 
         // Forward
-        ignore(self.send_swarm_or_parallel_or_relay(&message));
+        ignore(self.send_swarm_or_parallel_or_relay_with_signature(
+            &message, message_wrap.signature().clone()));
```

Comment:
I went for this approach - fully in agreement it is not the most elegant - as it is clear, correct and consistent with the approach where no destination is passed into the send_* function (from handlers side) as to make the destination always read from the RoutingMessage;

equally taking into account that no additional unwrapping of the RoutingMessage needs to occur.